### ISSUE TITLE
Port to GTK+ 3

### ISSUE THREAD

**Issue by ale-carra** [rickyrockrat/parcellite#21]
Will you port this to GTK+ 3? GTK+ 2 is dead.

**Comment by rickyrockrat** [comment]
GTK+3 was unstable for so long (and still has a lot of issues).  I will likely evaluate the possibilities and choose a GUI tool kit. FLTK is a strong possibility. Gnome pretty much killed itself with GTK+3.

**Comment by mooninite** [comment]
@rickyrockrat, Gnome Shell has dropped support for the legacy system tray. This means your app no longer does anything on Gnome 3.28. Updating the application to use GTK3 and the Gnome Shell notication area would make this application useful again.

I'm not sure where the comments about GTK3 come from, but they have no place here. You can either update the application or fade away into the sunset.

**Comment by sanjayankur31** [comment]
If moving to GTK3 is not an option, for whatever reason, I wonder if a shell extension would be in order then? 

https://blogs.gnome.org/aday/2017/08/31/status-icons-and-gnome/ -> This is worth a read, by the way.

**Comment by rickyrockrat** [comment]
There may be a possibility of moving to GTK3. I decided to take a look at seeing how much work it would be. I may build a branch and see how that fares. 
@mooninite - who are you to tell me what I can say on my project? Did you somehow achieve god-like status? Please be a thorn in someone else's project.

**Comment by felagund** [comment]
And Parcellite can be used without icon, so it can work great on Gnome too, I assume - I am still running it on Unity so not sure. Thanks for your work, Rickyrockrat!

**Comment by felagund** [comment]
But now with GTK 4 I am afraid GTK2 libreries will stop being included in distributions...

**Comment by johntyree** [comment]
The answer seems to be "stop using GTK."

This is a very simple app with a very simple UI. Migrating to a new UI framework every 2-3 years is a huge waste of time and resources.

**Comment by rickyrockrat** [comment]
johntyree - I think you might be right. This reminds me of an app I wrote a long time ago that works with imlib and is a layer right on top of X. I know folks using QT, but that has it's own issues as well. Perhaps it's time to remove all GUI toolkits and use something even lower level.

**Comment by johntyree** [comment]
There plenty of good, generic "menu" apps available for X11. I'd be fine with dzen2 or something, probably. At any rate, it seems like there's no need to roll your own if you don't enjoy it.

**Comment by martenwa** [comment]
@rickyrockrat I share your frustration with the toolkit updates, and it would indeed be nice in the future to remove those dependencies. This appears, however, at least to my naïve eyes, to be substantially more work than porting to GTK3 (or, as someone mentioned, GTK4). I am very satisfied with Parcellite, and I would be saddened to see it go when distributions finally stop shipping GTK2.

**Comment by p5n** [comment]
Arch started gtk2 drop but still has about 50 packages that depends on gtk2.

Also gtk2-compat package (https://archlinux.org/packages/community/any/gtk2-compat/) provided. It helps to compile some packages against gtk3 without modification but does not look that it helps parcellite.

**Comment by a-pav** [comment]
Through an upgrade I noticed that [debian maintainers have ported your project to gtk3](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=967681). Now it's visibly less snappy on the popup trigger (artha comes up faster), has excess padding between the entries and on the sides, and for some reason, it turns red when I use it with my own xkb bindings!

<img width="557" height="193" alt="Image" src="https://github.com/user-attachments/assets/c0783abd-68f7-454e-9883-89c0b142033e" />

I hope you could remedy these issues by an official solution in upstream.

**Comment by jkirk** [comment]
@a-pav I am currently working on the Debian package and try to clean up a few things... FTR, the GTK-3 port is not made by us, but was taken from here: https://github.com/rickyrockrat/parcellite/pull/43. We don't see red borders when using the hot-key. 

@rickyrockrat We just try to get the lastest version packaged for Debian/unstable. The latest https://github.com/rickyrockrat/parcellite/pull/121 to port parcellite to GTK-3 (which super seeds #43) works fine for us. We will test the lastest package with Wayland, but as far as we can say, the patch works fine.

**Comment by a-pav** [comment]
@jkirk Thanks for the work you're doing.

I was not talking about the hot-keys that we can set through the parcellite's preference window. I have a set of custom keybindings that are loaded by `xkbcomp`, they use `Hyper_R` as their modifier... Basically, parcellite becomes red at the press of `Hyper_R`, which is a modifier (like `Super_R`,`Shift_R` or `Control_R` etc.) so it shouldn't be considered a search character with the "Search as you type" feature, just like no other modifier is considered as such.

If you want to simulate what I'm talking about, enable the "Search as you type" feature, then use these:
```sh
xdotool key ctrl+alt+h sleep .5 key Super_R # (Or any other modifier except Hyper_R) OK, no red
xdotool key ctrl+alt+h sleep .5 key Hyper_R # RED!
```

As a side note, I think that parcellite could indicate the "no results" state for the "Search as you type" feature in a subtler way, but perhaps this should be discussed on a separate issue :) Right now, I have disabled this feature.

Since my previous comment, I've also changed the GTK theme I use, and I no longer see an issue with the padding/spacing between the entries, but the popup trigger is visibly a tad slower than it used to be.
